File: auto-dealer-sales-modern/frontend/src/types/customer.ts
// ──────────────────────────────────────────────
// Customer
// ──────────────────────────────────────────────

export interface Customer {
  customerId: number;
  firstName: string;
  lastName: string;
  middleInit: string | null;
  dateOfBirth: string | null;
  ssnLast4: string | null;
  driversLicense: string | null;
  dlState: string | null;
  addressLine1: string;
  addressLine2: string | null;
  city: string;
  stateCode: string;
  zipCode: string;
  homePhone: string | null;
  cellPhone: string | null;
  email: string | null;
  employerName: string | null;
  annualIncome: number | null;
  customerType: string;
  sourceCode: string | null;
  dealerCode: string;
  assignedSales: string | null;
  createdTs: string;
  updatedTs: string;
  formattedPhone: string | null;
  formattedCellPhone: string | null;
  fullName: string;
}

export interface CustomerRequest {
  firstName: string;
  lastName: string;
  middleInit?: string | null;
  dateOfBirth?: string | null;
  ssnLast4?: string | null;
  driversLicense?: string | null;
  dlState?: string | null;
  addressLine1: string;
  addressLine2?: string | null;
  city: string;
  stateCode: string;
  zipCode: string;
  homePhone?: string | null;
  cellPhone?: string | null;
  email?: string | null;
  employerName?: string | null;
  annualIncome?: number | null;
  customerType: string;
  sourceCode?: string | null;
  dealerCode: string;
  assignedSales?: string | null;
}

// ──────────────────────────────────────────────
// Customer History
// ──────────────────────────────────────────────

export interface CustomerHistory {
  customerId: number;
  customerName: string;
  repeatStatus: string;
  totalPurchases: number;
  totalSpent: number;
  averageDeal: number;
  deals: DealSummary[];
}

export interface DealSummary {
  dealNumber: string;
  dealDate: string;
  vin: string;
  yearMakeModel: string;
  dealType: string;
  salePrice: number;
  tradeAllow: number;
}

// ──────────────────────────────────────────────
// Credit Check
// ──────────────────────────────────────────────

export interface CreditCheckRequest {
  customerId: number;
  monthlyDebt?: number;
  bureauCode?: string;
}

export interface CreditCheckResponse {
  creditId: number;
  customerId: number;
  customerName: string;
  annualIncome: number;
  monthlyIncome: number;
  creditTier: string;
  creditTierDesc: string;
  creditScore: number;
  bureauCode: string;
  dtiRatio: number;
  monthlyDebt: number;
  maxFinancing: number;
  expiryDate: string;
  status: string;
  message: string;
}

// ──────────────────────────────────────────────
// Customer Lead
// ──────────────────────────────────────────────

export interface Lead {
  leadId: number;
  customerId: number;
  dealerCode: string;
  leadSource: string;
  interestModel: string | null;
  interestYear: number | null;
  leadStatus: string;
  assignedSales: string;
  followUpDate: string | null;
  lastContactDt: string | null;
  contactCount: number;
  notes: string | null;
  createdTs: string;
  updatedTs: string;
  customerName: string;
  overdue: boolean;
}

export interface LeadRequest {
  customerId: number;
  dealerCode: string;
  leadSource: string;
  interestModel?: string;
  interestYear?: number;
  assignedSales: string;
  followUpDate?: string;
  notes?: string;
}

File: auto-dealer-sales-modern/src/main/java/com/autosales/modules/customer/entity/Customer.java
package com.autosales.modules.customer.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "customer")
public class Customer {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "customer_id")
    private Integer customerId;

    @Column(name = "first_name", nullable = false, length = 30)
    private String firstName;

    @Column(name = "last_name", nullable = false, length = 30)
    private String lastName;

    @Column(name = "middle_init", length = 1)
    private String middleInit;

    @Column(name = "date_of_birth")
    private LocalDate dateOfBirth;

    @Column(name = "ssn_last4", length = 4)
    private String ssnLast4;

    @Column(name = "drivers_license", length = 20)
    private String driversLicense;

    @Column(name = "dl_state", length = 2)
    private String dlState;

    @Column(name = "address_line1", nullable = false, length = 50)
    private String addressLine1;

    @Column(name = "address_line2", length = 50)
    private String addressLine2;

    @Column(name = "city", nullable = false, length = 30)
    private String city;

    @Column(name = "state_code", nullable = false, length = 2)
    private String stateCode;

    @Column(name = "zip_code", nullable = false, length = 10)
    private String zipCode;

    @Column(name = "home_phone", length = 10)
    private String homePhone;

    @Column(name = "cell_phone", length = 10)
    private String cellPhone;

    @Column(name = "email", length = 60)
    private String email;

    @Column(name = "employer_name", length = 40)
    private String employerName;

    @Column(name = "annual_income", precision = 11, scale = 2)
    private BigDecimal annualIncome;

    @Column(name = "customer_type", nullable = false, length = 1)
    private String customerType;

    @Column(name = "source_code", length = 3)
    private String sourceCode;

    @Column(name = "dealer_code", nullable = false, length = 5)
    private String dealerCode;

    @Column(name = "assigned_sales", length = 8)
    private String assignedSales;

    @Column(name = "created_ts", nullable = false)
    private LocalDateTime createdTs;

    @Column(name = "updated_ts", nullable = false)
    private LocalDateTime updatedTs;
}
